modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.cc

```diff
@@ -118,41 +92,38 @@ LBSSolverIO::ReadFluxMoments(LBSSolver& lbs_solver,
 {
   // Open file
   const auto file_name =
-    file_base + (single_file ? "" : std::to_string(opensn::mpi_comm.rank())) + ".data";
-  std::ifstream file(file_name, std::ofstream::binary | std::ofstream::in);
-  OpenSnLogicalErrorIf(not file.is_open(), "Failed to open " + file_name + ".");
+    file_base + (single_file ? "" : std::to_string(opensn::mpi_comm.rank())) + ".h5";
+  hid_t file_id = H5Fopen(file_name.c_str(), H5F_ACC_RDONLY, H5P_DEFAULT);
+  OpenSnLogicalErrorIf(file_id < 0, "Failed to open " + file_name + ".");
 
   std::vector<double>& dest =
     opt_dest.has_value() ? opt_dest.value().get() : lbs_solver.PhiOldLocal();
 
   log.Log() << "Reading flux moments from " << file_base;
 
-  // Read the header
-  const int num_bytes = 500;
-  char header_bytes[num_bytes];
-  header_bytes[num_bytes - 1] = '\0';
-  file.read(header_bytes, num_bytes - 1);
-
-  // Read the macro info
+  // Read the macro data
   uint64_t file_num_local_cells;
+  H5ReadAttribute(file_id, "mesh/num_local_cells", file_num_local_cells);
+
   uint64_t file_num_local_nodes;
+  H5ReadAttribute(file_id, "mesh/num_local_nodes", file_num_local_nodes);
+
   uint64_t file_num_moments;
-  uint64_t file_num_groups;
+  H5ReadAttribute(file_id, "num_moments", file_num_moments);
 
-  file.read((char*)&file_num_local_cells, sizeof(uint64_t));
-  file.read((char*)&file_num_local_nodes, sizeof(uint64_t));
-  file.read((char*)&file_num_moments, sizeof(uint64_t));
-  file.read((char*)&file_num_groups, sizeof(uint64_t));
+  uint64_t file_num_groups;
+  H5ReadAttribute(file_id, "num_groups", file_num_groups);
 
   // Check compatibility with system macro info
+  const auto& grid = lbs_solver.Grid();
+  const auto& discretization = lbs_solver.SpatialDiscretization();
   const auto uk_man = lbs_solver.UnknownManager();
   const auto NODES_ONLY = UnknownManager::GetUnitaryUnknownManager();
```

Comment:
While you're refactoring, can you change `NODES_ONLY` so that it's not uppercase?